lib/screens/UsersProfileScreen.dart

```diff
@@ -0,0 +1,1576 @@
+                                      unfilledConfirmPassword = true;
+                                    });
+                                  }
+                                  else {
+                                    String newPassword = _newPassword.value.text.trim();
+                                    String changes = '{"firstname": "${user.firstName}","lastname": "${user.lastName}","lastSeen": ${user
+                                        .lastSeen}"username": "${user.email}","password": "$newPassword"}';
+                                    try {
+                                      final response = await User.updateUser(changes);
+
+                                      bool accepted = false;
+                                      while (!accepted) {
+                                        switch (response.statusCode) {
+                                          case 200:
+                                            accepted = true;
+                                            Navigator.pop(context);
+                                            break;
+                                          case 400:
+                                            print("Incorrect formatting!");
+                                            break;
+                                          case 401:
+                                            errorMessage =
+                                            'Access token missing';
+                                            break;
+                                          case 403:
+                                            final changeToken = await Authentication.refreshToken();
+                                            switch (changeToken.statusCode) {
+                                              case 200:
+                                                var tokens = json.decode(
+                                                    changeToken.body);
+                                                user.accessToken =
+                                                tokens['accessToken'];
+                                                user.refreshToken =
+                                                tokens['refreshToken'];
+                                                break;
+                                              case 400:
+                                                errorMessage =
+                                                'Could not refresh tokens';
+                                                break;
+                                              case 401:
+                                                errorMessage = 'Token missing';
+                                                break;
+                                              case 404:
+                                                errorMessage = 'User not found';
+                                                break;
+                                            }
```

Comment:
Since you do this in a couple places, you can have a dedicated function somewhere for refreshing the token/relogging in the user if the token expired; you're prob going to need that logic for every endpoint that requires the token. This can go in a helper util file.